packages/rum-core/src/performance-monitoring/performance-monitoring.js

```diff
@@ -372,58 +436,10 @@ class PerformanceMonitoring {
   }
 
   createTransactionPayload(transaction) {
-    this.prepareTransaction(transaction)
-    const filtered = this.filterTransaction(transaction)
+    const preparedTr = prepareTransaction(transaction)
```

Comment:
This could be a bit misleading since we still change the transaction itself, I don't think we should return the transaction but maybe we can change the function name to indicate it changes the transaction itself.

Reply:
Hmmm, yeah since deals mostly with the `spans`, we can call it as `adjustSpans` or `prepareSpans`?